src/core/expr/fexpr_sumprod.cc

```diff
@@ -97,7 +96,8 @@ class FExpr_SumProd : public FExpr_Func {
 
 
     template <typename T_IN>
-    Column make(Column &&col, const Groupby& gby, bool is_grouped) const {
+    Column make(Column &&col, SType stype, const Groupby& gby, bool is_grouped) const {
+      col.cast_inplace(stype);
```

Comment:
You need to restore it for all the reducers, as this PR is only related to `count()` function it is better to only change what is necessary.